tools/registry.py

```diff
@@ -434,6 +435,30 @@ def add(self, module, override=False):
     metadata["versions"].sort(key=Version)
     json_dump(metadata_path, metadata)
 
+  def update_versions(self, module_name):
+    """Update the list of versions in the metadata.json."""
+    metadata = self.get_metadata(module_name)
+    yanked_versions = set(metadata["yanked_versions"])
+    module_path = self.root / "modules" / module_name
+    actual_versions = {v.name for v in module_path.iterdir() if v.is_dir()}
+    metadata["versions"] = sorted(actual_versions - yanked_versions, key=Version)
+    metadata_path = self.get_metadata_path(module_name)
+    json_dump(metadata_path, metadata)
+
+  def update_integrity(self, module_name, version):
+    """Update the SRI hashes of the source.json file of module at version."""
+    source = self.get_source(module_name, version)
+    source["integrity"] = integrity(download(source["url"]))
+    source_path = self.get_source_path(module_name, version)
+    patch_dir = source_path.parent / "patches"
+    available = [p.name for p in patch_dir.iterdir()]
+    current = source.get("patches", {}).keys()
+    patch_files = [patch_dir / p for p in current if p in available]
+    patch_files.extend(patch_dir / p for p in available if p not in current)
```

Comment:
Could we sort the newly added patches?

Reply:
haven't checked but I'm pretty sure iterdir will give a sorted list already

Reply:
ah or do you mean insert them at the correct alphabetic order positions inside the "current" patches? I think this is good enough, and users can always edit afterwards :P

Reply:
Ah, no, just sort the newly added ones relative to each other to make the result deterministic. iterdir isn't documented to sort: https://docs.python.org/3/library/pathlib.html#pathlib.Path.iterdir

Reply:
indeed! done